### DISCUSSION TITLE
Compatible with React Native for version 0.15

### DISCUSSION THREAD

**Discussion by baochungit** [colyseus/colyseus#604] [General]
The version 0.14 was working very well on my React Native project however after upgraded to 0.15 it raised an error related to the URL class.

Here is my workaround for this issue:
```
import * as Url from 'url';

class NewURL extends URL {
  constructor(url, base) {
    super(url, base);
    this.parsed = Url.parse(url);
  }
  get hash() {
    return this.parsed.hash ?? '';
  }
  get host() {
    return this.parsed.host ?? '';
  }
  get hostname() {
    return this.parsed.hostname ?? '';
  }
  get pathname() {
    return this.parsed.pathname ?? '';
  }
  get port() {
    return this.parsed.port ?? '';
  }
  get protocol() {
    return this.parsed.protocol ?? '';
  }
  get search() {
    return this.parsed.search ?? '';
  }
}
window.URL = NewURL;
```
Do you guys intend to make it compatible with React Native in the future?

**Comment by endel** [comment]
Hi @baochungit, can you share the error (+stack trace) you have experienced?

**Reply by baochungit** [reply]
Not much info, just the error: `Error: URL.protocol is not implemented` and that leads to the error after `Invariant Violation: "main" has not been registered.` which prevents the app to start.

After a few minutes of searching, I realised that the error was raised at 'node_modules/react-native/Libraries/Blob/URL.js' file.
("react-native": "0.69.6")

**Reply by endel** [reply]
I see, thank you! 

We're using `new URL()` in the `Client`'s constructor in order to parse its `protocol`, `port`, `hostname`, and `pathname`: https://github.com/colyseus/colyseus.js/blob/5f39ab699aac9011f7945954c07fb149b109600f/src/Client.ts#L36-L48

The problem is that React Native does not provide a complete `URL` implementation - there is a polyfill available that can be used, though: https://github.com/charpeni/react-native-url-polyfill#option-1-simple
